### PR TITLE
Pol 222 rate exceeded aws unattached volumes

### PR DESCRIPTION
**Pull request by SheikSena**
### Description

This Policy Template scans all volumes in the given account and identifies any unattached volumes that have been unattached for at least the number of user-specified days. If any are found, an incident report will show the volumes, and related information and an email will be sent to the user-specified email address.

### Issues Resolved

All the minor changes have been done. But still, **we are facing ThrottlingException due to a large number of lookup events** 
### Contribution Check List

- [ ] All tests pass.
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [x] New functionality has been documented in CHANGELOG.MD

Link to check the changes : https://governance.rightscale.com/org/1105/projects/60073/policy-incidents/5e5ca3c9e69f0300012766fc

image

### CODE REVIEW

**Comment by SheikSena** [comment]
@cdwilhelm **This policy works only if the volume has less number of lookup events**. For AWS cloud trail logs API we have added a 1-second delay between 2 subsequent calls for different volume ids but if a single volume contains a large number of lookup events then we are facing ThrottlingException. Since it is a predefined pagination block we cannot delay the next request in the pagination block. Please let me know if there is any better approach to overcome this issue?

File: cost/aws/unattached_volumes/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## v3.0
```
Comment (cdwilhelm) [inline cost/aws/unattached_volumes/CHANGELOG.md]: This would be version 2.1.

File: cost/aws/unattached_volumes/aws_delete_unattached_volumes.pt [hunk 2]
```diff
@@ -5,7 +5,7 @@ short_description "Checks for unattached volumes older than specified number of
 category "Cost"
 severity "low"
 info(
-  version: "2.0",
+  version: "3.0",
```
Comment (cdwilhelm) [inline cost/aws/unattached_volumes/aws_delete_unattached_volumes.pt]: version should be 2.1

File: cost/aws/unattached_volumes/aws_delete_unattached_volumes.pt [hunk 3]
```diff
@@ -302,7 +303,7 @@ policy "policy_unattached_volumes_list" do
   summary_template "{{ rs_project_name }} (Account ID: {{ rs_project_id }}): {{ len data }} Unattached Volumes Found in AWS"
   detail_template <<-EOS
 # List of Unattached Volumes
-| Region | Volume_ID | Availability_Zones | Volume_Type | TAG_Value | Volume_UnAttached_Date |
+| Region | Volume Id | Availability Zones | Volume Type | TAG Value | Volume Age |
```
Comment (cdwilhelm) [inline cost/aws/unattached_volumes/aws_delete_unattached_volumes.pt]: consider sorting by Region, Age then AZ.  Probably useful to move Volume Age to the left.  
Region | Age | AZ then the rest.

Comment (cdwilhelm) [reply cost/aws/unattached_volumes/aws_delete_unattached_volumes.pt]: The Volume Age field is showing a timestamp, should be the age of the volume instead.

**Comment by SheikSena** [comment]
@cdwilhelm I have made all the required changes and also updated the READMe file. Please let me know if any changes are required. Thanks

File: cost/aws/unused_volumes/README.md [hunk 4]
```diff
@@ -1,30 +1,30 @@
-# AWS Unattached Volumes
+# AWS Unused Volumes
 
 ## What it does
 
-This Policy Template scans all volumes in the given account and identifies any unattached volumes that have been unattached for at least the number of user-specified days. If any are found, an incident report will show the volumes, and related information and an email will be sent to the user-specified email address.
+This Policy Template scans all volumes in the given account and identifies any unused volumes and if no read/writes operations performed within the number of user-specified days. If any are found, an incident report will show the volumes, and related information and an email will be sent to the user-specified email address. Cloud watch metrics are used to find the volume read/write operations
 
 If the user approves that the volumes should be deleted, the policy will delete the volumes.
```
Comment (cdwilhelm) [inline cost/aws/unused_volumes/README.md]: I know this is left over from the old policy, it doesn't read well.  Please consider :

The volume is deleted after user approval.

File: cost/aws/unused_volumes/README.md [hunk 5]
```diff
@@ -1,30 +1,30 @@
...
 If the volume is not able to be deleted, say, due to it being locked, the volume will be tagged to indicate the CloudException error that was received.
 If the issue causing the delete failure is removed, the next run of the policy will delete the volume.
-Note: The unattached volumes report will reflect the updated set of unattached volumes on the subsequent run.
+Note: The unused volumes report will reflect the updated set of unused volumes on the subsequent run.
```
Comment (cdwilhelm) [inline cost/aws/unused_volumes/README.md]: replace report with incident.

**Pull request merged**